functorch/_src/make_functional.py

```diff
@@ -310,7 +316,7 @@ def forward(self, params, *args, **kwargs):
             _swap_state(self.stateless_model, self.names_map, old_state)
 
 
-def make_functional(model: nn.Module):
+def make_functional(model: nn.Module, params_require_grad: bool = True):
```

Comment:
I have one main question about how this API works:
1. params_require_grad: should we make all parameters require grad, or should we mirror the state on the nn.Module (can a parameter on nn.Module have requires_grad=False) ?

Reply:
Yes, I agree, maybe this API can be confusing at this point. Probably, naming should reflect the fact that we could disable enabled grads but not enable disabled... Let me think about the naming.

Reply:
Agree that we can fix this with naming, especially since @vfdev-5 (correctly) only sets the params to `requires_grad(False)` and never the other way around. Maybe (or at least as a jumping off point),`disable_params_grad` (flipping the boolean's default)?

Reply:
Let's make it as `disable_params_grad`